app/src/ui/merge-branch/merge.tsx

```diff
@@ -67,10 +72,16 @@ export class Merge extends React.Component<IMergeProps, IMergeState> {
 
     const currentBranch = props.currentBranch
     const defaultBranch = props.defaultBranch
+    // use the preselected branch otherwise use the default branch if it's not currently checked out
```

Comment:
This comment makes me think we should extract this to a function and :memo: it rather than this complex group of ternaries.